### DISCUSSION TITLE
How do I read the codebase?

### DISCUSSION THREAD

**Discussion by vkc1vk** [sgl-project/sglang#1967] [General]
Hi I wanted to start contributing to SGLang, but I am new to inference codebases.

What is the suggested reading order for components in the codebase?
Instead of attempting to read portions of codebase, should I try to solve a beginner issue?
Thanks!

**Comment by merrymercy** [comment]
Yes, you can start from these tutorials https://sgl-project.github.io/ and trace the code base.
You can also start with some "good first issues"
https://github.com/sgl-project/sglang/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22
